hikari/impl/bot.py

```diff
@@ -543,18 +591,49 @@ def run(
             except AttributeError:
                 _LOGGER.debug("cannot set coroutine tracking depth for %s, no functionality exists for this", loop)
 
-        def signal_handler(signum: int) -> None:
-            raise errors.HikariInterrupt(signum, signal.strsignal(signum))
+        # Throwing this in the handler will lead to lots of fun OS specific shenanigans. So, lets just
+        # cache it for later, I guess.
+        interrupt: typing.Optional[errors.HikariInterrupt] = None
```

Comment:
You could possibly achieve this using a hack with `types.SimpleNamespace` if you wanted to move the handler but you'd still need a way to pass it around so I think this is fine here. It just looks bad because of my essay.